sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/OidcClientConfiguration.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package dev.sigstore.sign
+
+import org.gradle.api.Named
+
+interface OidcClientConfiguration : Named {
+    /**
+     * Creates OidcClient. The return type is [Any]
```

Comment:
Should we have an API lib that is exposed with all these interfaces?

Reply:
I do not think it is common for Gradle plugins to separate API interfaces to a separate jar. Pf course we could do that in the future, however, I belive it would complicate code without real gain at this point.

Reply:
oh sorry, I mean from sigstore-java

Reply:
Sigstore-java has too many dependecies, so adding a runtime dependency on sigstore-java is a hard no-go.

As you might notice, gradle plugin uses classloader isolation to isolate user's build scripts from sigstore-java dependencies

Reply:
Then, Named is Gradle interface, so adding dependency on Gradle to sigstore-java would be unwanted as well.

Reply:
sgtm. yeah the classpath isolation was nice.

Reply:
Ah, now I got what you mean. Of course it would be nice to have a small sigstore-java-api jar that exposes something like OidcClient without adding deps on Gson, etc, etc.

I just thought to fix up it later